src/libs/WorkspacesUtils.ts

```diff
@@ -0,0 +1,205 @@
+    const policyReports = policyID ? Object.values(allReports).filter((report) => report?.policyID === policyID) : Object.values(allReports);
+
+    let hasChatTabGBR = false;
+
+    const hasChatTabRBR = policyReports.some((report) => {
+        const brickRoad = report ? getBrickRoadForPolicy(report) : undefined;
+        if (!hasChatTabGBR && brickRoad === CONST.BRICK_ROAD_INDICATOR_STATUS.INFO) {
+            hasChatTabGBR = true;
+        }
+        return brickRoad === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+    });
+
+    if (hasChatTabRBR) {
+        return CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+    }
+
+    if (hasChatTabGBR) {
+        return CONST.BRICK_ROAD_INDICATOR_STATUS.INFO;
+    }
+
+    return undefined;
+}
+
+function checkIfWorkspaceSettingsTabHasRBR(policyID?: string) {
+    if (!policyID) {
+        return hasGlobalWorkspaceSettingsRBR(allPolicies, allPolicyMembers);
+    }
+    const policy = allPolicies ? allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`] : null;
+
+    if (!policy) {
+        return false;
+    }
+
+    return hasWorkspaceSettingsRBR(policy);
+}
+
+/**
+ * @returns a map where the keys are policyIDs and the values are BrickRoads for each policy
+ */
+function getWorkspacesBrickRoads(): Record<string, BrickRoad> {
+    if (!allReports) {
+        return {};
+    }
+
+    // The key in this map is the workspace id
+    const workspacesBrickRoadsMap: Record<string, BrickRoad> = {};
+
+    Object.values(allPolicies ?? {}).forEach((policy) => {
+        if (!policy) {
+            return;
+        }
+
+        if (hasWorkspaceSettingsRBR(policy)) {
+            workspacesBrickRoadsMap[policy.id] = CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+        }
+    });
+
+    Object.values(allReports).forEach((report) => {
+        const policyID = report?.policyID ?? CONST.POLICY.EMPTY;
+        if (!report || workspacesBrickRoadsMap[policyID] === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR) {
+            return;
+        }
+        const workspaceBrickRoad = getBrickRoadForPolicy(report);
+
+        if (!workspaceBrickRoad && !!workspacesBrickRoadsMap[policyID]) {
+            return;
+        }
+
+        workspacesBrickRoadsMap[policyID] = workspaceBrickRoad;
+    });
+
+    return workspacesBrickRoadsMap;
+}
+
+/**
+ * @returns a map where the keys are policyIDs and the values are truthy booleans if policy has unread content
+ */
+function getWorkspacesUnreadStatuses(): Record<string, boolean> {
+    if (!allReports) {
+        return {};
+    }
+
+    const workspacesUnreadStatuses: Record<string, boolean> = {};
+
+    Object.values(allReports).forEach((report) => {
+        const policyID = report?.policyID;
+        if (!policyID || workspacesUnreadStatuses[policyID]) {
+            return;
+        }
+
+        workspacesUnreadStatuses[policyID] = ReportUtils.isUnread(report);
+    });
+
+    return workspacesUnreadStatuses;
+}
```

Comment:
@mountiny @s77rt I've modified this function and optimized it a bit, but I don't see a case to use `some` here. Do you think it's ok or could it be optimized better?

Reply:
I think this works well. the other option would be to iterate over the policies and use `some` for ReportUtils.isUnread(report) on that policy.

Bu I think what we have now might be even better!